src/confluent_kafka/src/AdminTypes.c

```diff
@@ -125,7 +127,8 @@ static PyMemberDef NewTopic_members[] = {
         { "topic", T_STRING, offsetof(NewTopic, topic), READONLY,
           ":py:attribute:topic - Topic name (string)" },
         { "num_partitions", T_INT, offsetof(NewTopic, num_partitions), 0,
-          ":py:attribute: Number of partitions (int)" },
+          ":py:attribute: Number of partitions (int).\n"
+          "Must be set to -1 if a replica_assignment is specified" },
```

Comment:
If we are enforcing this, then we don't need this condition.

```
if (newt->num_partitions == -1) {
        t = PyList_Size(newt->replica_assignment);
} else {
        t = newt->num_partitions;
}
```

We can use what's written in Java. In this way, its a suggestion.

> The number of partitions for the new topic or -1 if a replica assignment has been specified.

Reply:
We are not enforcing to keep backward compatibility, we need the check so prior code does not break.